### PR TITLE
ARROW-16949: [Doc] Add Glossary to the New Contributor's Guide

### PR DESCRIPTION
**Pull request by AlenkaF**

### CODE REVIEW

File: docs/source/developers/guide/resources.rst [hunk 1]
```diff
@@ -31,9 +31,10 @@ Additional information and resources
 On this page we have listed resources that may be relevant or useful for
 contributors who want to learn more about different parts of Apache Arrow.
 
-.. Annotation Glossary
-.. -------------------
-.. ARROW-15130
+Annotation Glossary
```
Comment (lidavidm) [inline docs/source/developers/guide/resources.rst:34]: just a drive-by nit: what is the 'annotation' part here referring to?

Comment (AlenkaF) [reply docs/source/developers/guide/resources.rst:34]: To the short description of the term in the Glossary. It is mainly left from when searching for the projects having the glossary and saw they are naming it Annotation Glossary. Will remove it so it is in sync with [ARROW-15130](https://issues.apache.org/jira/browse/ARROW-15130) 👍

**Review by raulcd — APPROVED** [review]
Thanks for this @AlenkaF

**Review by lidavidm — APPROVED** [review]
LGTM.

(Though note that `:doc:` lets you link a page without having to add an anchor.)

**Comment by AlenkaF** [comment]
Thanks @lidavidm! Will try it out, commit the change and then merge the PR 👍
